Library/Homebrew/build.rb

```diff
@@ -190,7 +190,20 @@ def fixopt(f)
   build   = Build.new(formula, options)
   build.install
 rescue Exception => e # rubocop:disable Lint/RescueException
-  Marshal.dump(e, error_pipe)
+  error_hash = JSON.parse e.to_json
+
+  # Special case: need to recreate BuildErrors in full
+  # for proper analytics reporting and error messages.
+  # BuildErrors are specific to build processses and not other
+  # children, which is why we create the necessary state here
+  # and not in Utils.safe_fork.
+  if e.is_a?(BuildError)
```

Comment:
Is there any way to avoid all the `is_a?` Maybe looking at the keys instead?

Reply:
Yeah, `error_hash["json_class"] == "BuildError"` should do the trick. I'll change it.